code/modules/surgery/organs/organ_internal.dm

```diff
@@ -343,16 +343,41 @@
 	parent_organ = "head"
 	slot = "eyes"
 	var/list/eye_colour = list(0,0,0)
+	var/list/colourmatrix = null
+	var/list/colourblind_matrix = null //Special colourblindness parameters.
+	var/colourblind_darkview = null
+	var/dependent_disabilities = null //Gets set by eye-dependent disabilities such as colourblindness so the eyes can transfer the disability during transplantation.
+	var/dark_view = 2 //Default dark_view for Humans.
+	var/weld_proof = null //If set, the eyes will not take damage during welding. eg. IPC optical sensors do not take damage when they weld things while all other eyes will.
 
 /obj/item/organ/internal/eyes/proc/update_colour()
 	dna.write_eyes_attributes(src)
 
-/obj/item/organ/internal/eyes/insert(mob/living/carbon/M, special = 0)
+/obj/item/organ/internal/eyes/proc/get_colourmatrix() //Returns a colour matrix if the eyes are organic.
+	return !robotic && owner.disabilities & COLOURBLIND ? (colourblind_matrix ? colourblind_matrix : MATRIX_GREYSCALE) : colourmatrix //Returns special colourmatrix if colourblind and it exists and greyscale if it doesn't, otherwise returns current velue.
+
+/obj/item/organ/internal/eyes/proc/get_dark_view() //Returns dark_view (if the eyes are organic) for see_invisible handling in species.dm to be autoprocessed by life().
+	return !robotic && colourblind_darkview && owner.disabilities & COLOURBLIND ? colourblind_darkview : dark_view //Returns special darkview value if colourblind and it exists, otherwise returns current velue.
```

Comment:
Lol, not really. I thought it was sick I was able to do what would've been an `if` and a couple lines in-line with the `return` because of the ? operator so I decided to use it.

But yeah your above points are valid, I'll switch it up